Index: src/com/company/Game/Board.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/com/company/Game/Board.java	(date 1546970539078)
+++ src/com/company/Game/Board.java	(date 1546970539078)
@@ -2,7 +2,12 @@
 
 import com.company.Tiles.*;
 
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
 import java.util.*;
+import java.util.stream.Collectors;
 /**
  * @authors icsd15058 ZIOZAS GEORGIOS
  * icsd15213 CHALAGIANNIS XRISTOS
@@ -36,8 +41,9 @@
                     "1m", "2m", "3m", "4m", "5m", "6m", "7m", "8m", "9m",
                     "1e", "2e", "3e", "4e",
                     "1e", "2e", "3e", "4e",
-                    "1l", "2l", "3l", "4l" ,
-                    "1l", "2l", "3l", "4l"};
+                    "1l", "2l", "3l", "4l",
+                    "1l", "2l", "3l", "4l"
+            };
 
 
 
@@ -50,7 +56,7 @@
      * In this lists we will copy the elements of the String Arrays(pXaraktiron,Pkiklon...etc) so we can add them as lines in the board 2d List)
      */
     public List<String> tilesList;
-
+    public List<String> list;
 
     /**
      * Board, Constructor of the class.
@@ -60,18 +66,14 @@
         System.out.println(getBoard());
     }
 
-
     public static List<List<Tiles>> getBoard() {
         return board;
     }
-
-
     /**
      * @return void - Optional sout at the end for testing.
      * @desc Initializes all the Tiles objects to the board array.
      */
 
-
     public void BoardInitialiaztion() {
         Random rand = new Random(10);
         int emptyBtns = 5;
@@ -83,6 +85,7 @@
         Collections.shuffle(tilesList, rand);
         System.out.println(tilesList);
         int ctr_listTiles = 0;
+
         board = new ArrayList<>(boardSize);     //initializing the board dynamically cause the board doesnt have same num of rows. With capacity = boardsize (124).
 
 
@@ -131,20 +134,67 @@
 //        }*/
 
 
-        for (int i = 0; i < 13; i++)
-        {
-            board.add(new ArrayList<>(9));
-            for (int j = 0; j < 9; j++)
-            {
-                if (tilesList.get(j).contains("x")) { board.get(i).add(new Xaraktiron(tilesList.get(ctr_listTiles), new int[]{i, j}, "blue"));}
-                if (tilesList.get(j).contains("k")) { board.get(i).add(new Kiklon(tilesList.get(ctr_listTiles), new int[]{i, j}, "blue")); }
-                if (tilesList.get(j).contains("m")) { board.get(i).add(new Mpampou(tilesList.get(ctr_listTiles), new int[]{i, j}, "blue")); }
-                if (tilesList.get(j).contains("l")) { board.get(i).add(new Louloudion(tilesList.get(ctr_listTiles), new int[]{i, j}, "blue")); }
-                if (tilesList.get(j).contains("e")) { board.get(i).add(new Epoxon(tilesList.get(ctr_listTiles), new int[]{i, j}, "blue")); }
-                else { }
-                ctr_listTiles++;
+
+
+        list = new ArrayList<>();
+
+        try (BufferedReader br = new BufferedReader(new FileReader("level1.txt"))) {
+
+            list = br.lines().collect(Collectors.toList());
+            System.out.println(list);
+
+            for (int i = 0; i < list.size(); i++)      //list.size()=14
+            {
+                board.add(new ArrayList<>(9));
+
+                for (String temp : list) {
+                    int tempj = 0;
+                    if (temp == "0") {
+                        board.get(i).add(new emptyBlock("empty", new int[]{0, 0}, "b"));
+                        tempj++;
+                    }
+
+                    else if (temp == "1") {
+
+                        if (ctr_listTiles < 124)   //oso iparxoun xaraktires akoma p prepei na mpoun sto pinaka
+                        {
+                            if (tilesList.get(ctr_listTiles).contains("x")) {
+                                board.get(i).add(new Xaraktiron(tilesList.get(ctr_listTiles), new int[]{i, tempj}, "blue"));
+                            }
+                            else if (tilesList.get(ctr_listTiles).contains("k")) {
+                                board.get(i).add(new Kiklon(tilesList.get(ctr_listTiles), new int[]{i, tempj}, "blue"));
+                            }
+                            else if (tilesList.get(ctr_listTiles).contains("m")) {
+                                board.get(i).add(new Mpampou(tilesList.get(ctr_listTiles), new int[]{i, tempj}, "blue"));
+                            }
+                            else if (tilesList.get(ctr_listTiles).contains("l")) {
+                                board.get(i).add(new Louloudion(tilesList.get(ctr_listTiles), new int[]{i, tempj}, "blue"));
+                            }
+                            else if (tilesList.get(ctr_listTiles).contains("e")) {
+                                board.get(i).add(new Epoxon(tilesList.get(ctr_listTiles), new int[]{i, tempj}, "blue"));
+                            }
+                            else {
+                            }
+                            tempj++;
+                        }
+                        ctr_listTiles++;
+
+                    }
+                }
+
             }
+
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
         }
+    }
+
+
+
+
+
 
 //      /*  for (int i = 14; i < 15; i++) {
 //            board.add(new ArrayList<>(9));
@@ -188,4 +238,4 @@
 
 
     }
-}
+
